lib/pure/httpclient.nim

```diff
@@ -890,16 +891,32 @@ proc parseResponse(client: HttpClient | AsyncHttpClient,
       parsedStatus = true
     else:
       # Parse headers
-      var name = ""
-      var le = parseUntil(line, name, ':', linei)
-      if le <= 0: httpError("invalid headers")
-      inc(linei, le)
-      if line[linei] != ':': httpError("invalid headers")
-      inc(linei) # Skip :
-
-      result.headers.add(name, line[linei .. ^1].strip())
-      if result.headers.len > headerLimit:
-        httpError("too many headers")
+      # There's at least one char because empty lines are handled above (with client.close)
+      if line[0] in {' ', '\t'}:
+        # Check if it's a multiline header value, if so, append to the header we're currently parsing
+        # This works because a line with a header must start with the header name without any leading space
+        # See https://datatracker.ietf.org/doc/html/rfc7230, section 3.2 and 3.2.4
+        # Multiline headers are deprecated in the spec, but it's better to parse them than crash
+        if lastHeaderName == "":
+          # This should ideally throw an error, but old httpclient actually treated lines like this
+          # as a header with an empty name and didn't crash
+          #httpError("Invalid headers - received multiline header value without previous header")
+          discard
```

Comment:
Should I change this or not? I just fear that some extremely weird servers might actually send some unparsed lines in the response, and httpclient before just ignored those (added to the empty header)

Reply:
FWIW, Python's requests also ignores this.

Reply:
Follow Python's requests then.